website/addons/osfstorage/decorators.py

```diff
@@ -0,0 +1,77 @@
+import httplib
+import functools
+
+from modularodm.exceptions import NoResultsFound
+from modularodm.storage.base import KeyExistsException
+from framework.auth.decorators import must_be_signed
+
+from framework.exceptions import HTTPError
+
+from website.models import User
+from website.models import Node
+from website.addons.osfstorage import model
+from website.project.decorators import (
+    must_not_be_registration, must_have_addon,
+)
+
+
+def handle_odm_errors(func):
+    @functools.wraps(func)
+    def wrapped(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except NoResultsFound:
+            raise HTTPError(httplib.NOT_FOUND)
+        except KeyExistsException:
+            raise HTTPError(httplib.CONFLICT)
+    return wrapped
+
+
+def autoload_filenode(must_be=None, default_root=False):
+    def _autoload_filenode(func):
+        @handle_odm_errors
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+            node_addon = kwargs['node_addon']
```

Comment:
Can you add documentation to indicate that this decorator must be used together with a decorator that injects `node_addon`?